README.md

```diff
@@ -1,52 +1,209 @@
+  - [Post JPEG image](#post-jpeg-image)
+  - [Post form data](#post-form-data)
+  - [GET with query parameters object](#get-with-query-parameters-object)
+- [Advanced Usage Examples](#advanced-usage-examples)
+  - [HTTP/2 Server Push](#http2-server-push)
+  - [Force HTTP/1(.1) protocol](#force-http11-protocol)
+  - [HTTP/1.1 Keep-Alive](#http11-keep-alive)
+  - [Self-signed Certificates](#self-signed-certificates)
+  - [Set cache size limit](#set-cache-size-limit)
+  - [Set a custom user agent](#set-a-custom-user-agent)
+- [More examples](#more-examples)
+- [Development](#development)
+  - [Build](#build)
+  - [Test](#test)
+  - [Lint](#lint)
+- [Acknowledgement](#acknowledgement)
+- [License](#license)
+<!-- /TOC -->
+
+---
+
+## About
+
+`helix-fetch` in general adheres to the [Fetch API Specification](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API), implementing a subset of the API. However, there are some notable deviations:
+
+* `Response.body` returns a Node.js [Readable stream](https://nodejs.org/api/stream.html#stream_readable_streams).
 * `Response.blob()` is not implemented. Use `Response.buffer()` instead.
 * `Response.formData()` is not implemented.
-* The following `fetch()` options are ignored since `helix-fetch` doesn't have the concept of web pages: `mode`, `referrer` and `referrerPolicy`.
+* Cookies are not stored by default. However, cookies can be extracted and passed by manipulating request and response headers.
+* The following `fetch()` options are ignored due to the nature of Node.js and since `helix-fetch` doesn't have the concept of web pages: `mode`, `referrer`, `referrerPolicy` `integrity` and `credentials`.
+* The `fetch()` option `keepalive` is not supported. But you can use the `h1.keepAlive` context option, as demonstrated [here](#http11-keep-alive).
 
 `helix-fetch` also supports the following extensions:
 
 * `Response.buffer()` returns a Node.js `Buffer`.
-* The `body` that can be sent in a `Request` can also be a `Readable` Node.js stream, a `Buffer` or a string.
-* `fetch()` has an extra option `json` that can be used instead of `body` to send an object that will be JSON stringified. The appropriate content-type will be set if it isn't already.
-* ~~`fetch()` has an extra option `timeout` which is a timeout in milliseconds before the request should be aborted and the returned promise thereby rejected (with a `TimeoutError`).~~
+* `Response.url` contains the final url when following redirects.
+* The `body` that can be sent in a `Request` can also be a `Readable` Node.js stream, a `Buffer`, a string or a plain object.
+* There are no forbidden header names.
+* The `Response` object has an extra property `httpVersion` which is one of `'1.0'`, `'1.1'` or `'2.0'`, depending on what was negotiated with the server.
+* The `Response` object has an extra property `fromCache` which determines whether the response was retrieved from cache.
+* `Response.headers.plain()` returns the headers as a plain object.
```

Comment:
why not `raw()` like node-fetch?  https://github.com/node-fetch/node-fetch/blob/4abbfd231f4bce7dbe65e060a6323fc6917fd6d9/src/headers.js#L200

Reply:
That's exactly the reason for renaming it to `plain()` ;-) `node-fetch`'s `raw()` method returns all headers and their values as array whereas `plain()` returns the headers as a plain JS object.

Reply:
a, right. thanks.